frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/contextmenu/ResetSelectedBranchToRemoteAction.java

```diff
@@ -0,0 +1,31 @@
+package com.virtuslab.gitmachete.frontend.actions.contextmenu;
+
+import static com.virtuslab.gitmachete.frontend.actions.common.ActionUtils.getSelectedBranchName;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import io.vavr.control.Option;
+
+import com.virtuslab.gitmachete.frontend.actions.common.BaseResetBranchToRemoteAction;
+import com.virtuslab.gitmachete.frontend.datakeys.DataKeys;
+
+/**
+ * Expects DataKeys:
+ * <ul>
+ *  <li>{@link DataKeys#KEY_GIT_MACHETE_REPOSITORY}</li>
+ *  <li>{@link DataKeys#KEY_SELECTED_BRANCH_NAME}</li>
+ *  <li>{@link DataKeys#KEY_SELECTED_VCS_REPOSITORY}</li>
+ *  <li>{@link CommonDataKeys#PROJECT}</li>
+ * </ul>
+ */
+public class ResetSelectedBranchToRemoteAction extends BaseResetBranchToRemoteAction {
+  @Override
+  public Option<String> getNameOfBranchUnderAction(AnActionEvent anActionEvent) {
+    return getSelectedBranchName(anActionEvent);
+  }
+
+  @Override
+  protected String getActionType() {
```

Comment:
we dont need that, see comment above (or below, idk, reviewing upwards XD)

Reply:
as above